nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/OperationContext.java

```diff
@@ -144,37 +144,57 @@ public Object getInput() {
         return input;
     }
 
-    public Object peek(String type) {
-        List<Object> stack = stacks.get(type);
-        if (stack == null) {
-            return null;
-        }
-        return stack.isEmpty() ? null : stack.get(stack.size() - 1);
+    public void push(Map<String, ?> map) {
+        map.forEach(this::push);
+    }
+
+    public void pop(Map<String, ?> map) {
```

Comment:
I find it surprising to have a `pop` method taking a `Map` and just using its keys. Please add a Javadoc.